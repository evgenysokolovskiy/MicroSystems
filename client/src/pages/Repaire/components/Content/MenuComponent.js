import React from 'react'
// Antd
import { Layout, Menu, Icon } from 'antd'
const { Sider } = Layout
const { SubMenu } = Menu

export const MenuComponent = () => {
    return (
        <Sider width={250} style={{ background: '#fff' }}>
            <Menu mode="inline" style={{ height: '100%' }}>
                <SubMenu key="sub1" title={<span> Производство №50</span>}>
                    <Menu.Item key="1">
                        <Icon type="solution" />
                        План ремонтов
                    </Menu.Item>
                    <Menu.Item key="2">
                        <Icon type="tool" />
                        Диагностика
                    </Menu.Item>
                    <Menu.Item key="3">
                        <Icon type="table" />
                        Схема
                    </Menu.Item>
                </SubMenu>

                <SubMenu key="sub2" title={<span>Производство №56</span>}>
                    <Menu.Item key="4">
                        <Icon type="solution" />
                        План ремонтов
                    </Menu.Item>
                    <Menu.Item key="5">
                        <Icon type="tool" />
                        Диагностика
                    </Menu.Item>
                    <Menu.Item key="6">
                        <Icon type="table" />
                        Схема
                    </Menu.Item>
                </SubMenu>

                <SubMenu key="sub3" title={<span>Производство №57</span>}>
                    <Menu.Item key="7">
                        <Icon type="solution" />
                        План ремонтов
                    </Menu.Item>
                    <Menu.Item key="8">
                        <Icon type="tool" />
                        Диагностика
                    </Menu.Item>
                    <Menu.Item key="9">
                        <Icon type="table" />
                        Схема
                    </Menu.Item>
                </SubMenu>

                <SubMenu key="sub4" title={<span>Производство №61</span>}>
                    <Menu.Item key="10">
                        <Icon type="solution" />
                        План ремонтов
                    </Menu.Item>
                    <Menu.Item key="11">
                        <Icon type="tool" />
                        Диагностика
                    </Menu.Item>
                    <Menu.Item key="12">
                        <Icon type="table" />
                        Схема
                    </Menu.Item>
                </SubMenu>

                <SubMenu key="sub5" title={<span>Производство №63</span>}>
                    <Menu.Item key="13">
                        <Icon type="solution" />
                        План ремонтов
                    </Menu.Item>
                    <Menu.Item key="14">
                        <Icon type="tool" />
                        Диагностика
                    </Menu.Item>
                    <Menu.Item key="15">
                        <Icon type="table" />
                        Схема
                    </Menu.Item>
                </SubMenu>
            </Menu>
        </Sider>
    )
}
