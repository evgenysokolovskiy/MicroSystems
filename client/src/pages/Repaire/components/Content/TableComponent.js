import React from 'react'
// Antd
import { Table, Tabs } from 'antd'
import { data, columns } from '../../config'
const { TabPane } = Tabs

export const TableComponent = props => {
    const { data } = props

    // Найти 'column'
    // Исходный вид
    let column = [
        {
            title: '№ п/п',
            dataIndex: 'idx',
            width: 60,
            fixed: 'left'
        },
        {
            title: 'Модель',
            dataIndex: 'model',
            width: 150,
            fixed: 'left',
            onFilter: (value, record) => record.model.indexOf(value) === 0,
            sorter: (a, b) => a.model.length - b.model.length,
            sortDirections: ['descend']
        },
        {
            title: 'Цех. №',
            dataIndex: 'num',
            width: 80,
            fixed: 'left',
            sorter: (a, b) => a.num - b.num,
            sortDirections: ['descend', 'ascend']
        },
        {
            title: 'Инв. №',
            dataIndex: 'inn',
            width: 90,
            fixed: 'left',
            sorter: (a, b) => a.inn - b.inn,
            sortDirections: ['descend', 'ascend']
        },
        {
            title: 'Вид',
            dataIndex: 'typeOfRepair',
            width: 90,
            fixed: 'left',
            sorter: (a, b) => a.typeOfRepair - b.typeOfRepair,
            sortDirections: ['descend', 'ascend']
        }
    ]

    // Добавить узлы
    let nodes = []
    let allNodes = {}
    if (data) {
        Object.keys(data['50']['nodes']).forEach(node => {
            allNodes[node] = ''
            const obj = {
                title: node,
                dataIndex: node,
                render(text) {
                    return {
                        props: {
                            className: text === ' ' ? 'color' : ''
                        },
                        children: text
                    }
                },
                width: 65,
                filters: [
                    {
                        text: `Имеет неисправность ${node}`,
                        value: ' '
                    },
                    {
                        text: 'Нет данной неисправности',
                        value: ''
                    }
                ],
                // Фильтровать данные
                onFilter: (value, record) => record[node].indexOf(value) === 0,
                // Событие на ячейке
                onCell: (record, rowIndex) => {
                    return {
                        onClick: event => {
                            console.log(event.currentTarget, rowIndex)
                        }, // click row
                        onDoubleClick: event => {}, // double click row
                        onContextMenu: event => {}, // right button click row
                        onMouseEnter: event => {}, // mouse enter row
                        onMouseLeave: event => {} // mouse leave row
                    }
                }
            }

            nodes = [...nodes, obj]
        })
    }

    column = [...column, ...nodes]

    let tabsWithTables = [] // Компонент для рендеринга

    if (data) {
        // Для каждого периода времени строится своя таблица
        // Далее соотносится с Tabs
        let tables = []
        data['50']['data'].forEach((period, index) => {
            // Найти 'dataSource'
            let dataSource = []
            period.forEach((item, i) => {
                const obj = {
                    key: i,
                    idx: ++i,
                    model: item['model'],
                    num: item['num'],
                    inn: item['inn'],
                    typeOfRepair: (() =>
                        item['typeOfRepair'] === 'medium' ? 'средний' : 'текущий')(),
                    ...allNodes
                }

                Object.keys(item['nodes']).forEach(node => {
                    obj[node] = ' '
                })

                dataSource = [...dataSource, obj]
            })

            // * Построить таблицу на основе column и datasource
            const table = (
                <>
                    <Table
                        columns={column}
                        dataSource={dataSource}
                        bordered
                        onChange={onChange}
                        pagination={false}
                        scroll={{ x: '10vw', y: '60vh' }}
                        size="small"
                        // Событие на строке
                        onRow={(record, rowIndex) => {
                            return {
                                onClick: event => {
                                    console.log(event.currentTarget, rowIndex)
                                }, // click row
                                onDoubleClick: event => {}, // double click row
                                onContextMenu: event => {}, // right button click row
                                onMouseEnter: event => {}, // mouse enter row
                                onMouseLeave: event => {} // mouse leave row
                            }
                        }}
                    />
                </>
            )
            tables = [...tables, table]
        })

        data['50']['period'].forEach(
            (item, i) =>
                (tabsWithTables = [
                    ...tabsWithTables,
                    <TabPane tab={item} key={i}>
                        {tables[i]}
                    </TabPane>
                ])
        )

        tabsWithTables = [...tabsWithTables]
    }

    return (
        <>
            <Tabs defaultActiveKey="1" type="card">
                {tabsWithTables}
            </Tabs>
        </>
    )
}

function onChange(pagination, filters, sorter, extra) {
    console.log('params', pagination, filters, sorter, extra)
}

/*


{
    
    50: {
        data: [
            [
                {
                    spot: 50,
                    model: '1kjkm',
                    inn: 123456789,
                    num: 123456,
                    ps: "",
                    mtbf: 12345,
                    nodes: {
                        1.02: {
                            time: '6,25',
                            amount: 17,
                            oneRepairTime: 0.12345,
                            percentOfAllNodes: 0.12,
                            repairComplexityNode: 10.8
                        },
                        1.03: {...},
                        2.1: {...}
                    },
                    sumAmount: 373,
                    sumTime: 100.42,
                    sumOneRepairTimeMechanic: 2.9964509025450203
                    sumOneRepairTimeElectric: 1.0087518207282913
                    lengthNodesMechanic: 9
                    lengthNodesElectric: 4
                    typeOfRepair: "medium"
                    mechanicRepairComplexity: 49
                    electricRepairComplexity: 51
                }
            ],
            [...],
            [...]
        ],
        allEquipment: 427
        filteredEquipment: 180
        middleCount: 92
        nodesCount: 88
        nodes: {1.03: {…}, 2.1: {…}, 1.1: {…}, 2.12: {…}, 1.08: {…}, …}
        sumAmountAllNodes: 2350
        sumTimeAllNodes: 583.1800000000002
        sumMechanicRepairComplexity: 4565.4
        sumElectricRepairComplexity: 7253.599999999998
        inPlanningPeriodMechanicRepairComplexity: 380.45,
        period: (12) ["Январь", "Февраль", "Март", "Апрель", "Май", "Июнь", "Ию"]
    }
}



*/
