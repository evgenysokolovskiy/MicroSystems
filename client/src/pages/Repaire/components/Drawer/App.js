import React from 'react'
import { Drawer } from 'antd'
import ChartComponent from '../Content/ChartComponent'

export const App = props => {
    const { visible, handleClickCloseDrawer } = props

    return (
        <div>
            <Drawer
                width={'50vw'}
                title="Basic Drawer"
                placement="right"
                closable={false}
                onClose={handleClickCloseDrawer}
                visible={visible}
            >
                <ChartComponent />
            </Drawer>
        </div>
    )
}
