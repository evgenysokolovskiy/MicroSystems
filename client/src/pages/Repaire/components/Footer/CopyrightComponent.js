import React from 'react'

export const CopyrightComponent = () => {
    return <span>Â©2019 Created by Sokolovsky Evgeny</span>
}
