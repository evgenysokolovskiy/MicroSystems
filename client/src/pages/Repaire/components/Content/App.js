import React from 'react'
// Компоненты
import { BreadcrumbComponent } from './BreadcrumbComponent'
import { MenuComponent } from './MenuComponent'
import { TableComponent } from './TableComponent'
// Antd
import { Layout } from 'antd'
const { Content } = Layout

export const App = () => {
    return (
        <Content style={{ padding: '0 50px' }}>
            <BreadcrumbComponent />
            <Layout style={{ padding: '24px 0', background: '#fff' }}>
                <MenuComponent />
                <Content style={{ padding: '0 24px', minHeight: 280 }}>
                    <TableComponent />
                </Content>
            </Layout>
        </Content>
    )
}
