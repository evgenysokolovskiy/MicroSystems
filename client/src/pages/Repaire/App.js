import React from 'react'
// Antd
import { Layout } from 'antd'
// Компоненты
import Header from './containers/Header'
import Content from './containers/Content'
import Footer from './containers/Footer'
// Стили
import './styles/index.css'

export default function Repaire() {
    return (
        <Layout>
            <Header />
            <Content />
            <Footer />
        </Layout>
    )
}
