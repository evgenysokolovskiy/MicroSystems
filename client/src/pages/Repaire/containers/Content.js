import React from 'react'
import { connect } from 'react-redux'
// Компонент
import { App } from '../components/Content/App'

export class Content extends React.Component {
    render() {
        console.log(this.props.data)
        return (
            <>
                <App />
            </>
        )
    }
}

function mapStateToProps(store) {
    return {
        ...store.fetchReducer
    }
}

const mapDispatchToProps = {}

export default connect(mapStateToProps, mapDispatchToProps)(Content)
