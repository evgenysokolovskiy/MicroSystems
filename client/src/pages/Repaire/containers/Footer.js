import React from 'react'
import { connect } from 'react-redux'
// Компоненты
import { App } from '../components/Footer/App'

export class Footer extends React.Component {
    render() {
        return (
            <>
                <App />
            </>
        )
    }
}

function mapStateToProps(store) {
    return {
        ...store.fetchReducer
    }
}

const mapDispatchToProps = {}

export default connect(mapStateToProps, mapDispatchToProps)(Footer)
