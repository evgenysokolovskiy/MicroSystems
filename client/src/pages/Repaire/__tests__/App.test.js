import React from 'react'
import { shallow } from 'enzyme'
import App from '../App'
import Header from '../containers/Header'
import Content from '../containers/Content'
import Footer from '../containers/Footer'

it('render correctly App', () => {
    const wrapper = shallow(<App />)
    expect(wrapper.contains(<Header />)).toEqual(true)
    expect(wrapper.contains(<Content />)).toEqual(true)
    expect(wrapper.contains(<Footer />)).toEqual(true)
    expect(wrapper).toMatchSnapshot()
})
