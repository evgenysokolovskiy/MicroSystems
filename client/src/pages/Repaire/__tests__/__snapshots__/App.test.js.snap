// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render correctly App 1`] = `ShallowWrapper {}`;
